Python assignment for uni Ml Course in Python

Import necessary packages

In [111]:
import pandas as pd

1. Read file titanic

In [112]:
df = pd.read_csv('C:/Users/Marc/Dropbox/06_ESCP/01_Uni/04_Term 2/03_Python ML/02_2nd Lecture/titanic.csv')
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

look at df closer

In [113]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

Q1: How many people are in titanic and how many survivors?

In [114]:
PassengerAmount = str(df.PassengerId.count())
SurvivedAmount = str(df.Survived.sum())


print('on the titanic there were ' + PassengerAmount + ' Passengers and \n' + SurvivedAmount + ' of them survived')

on the titanic there were 891 Passengers and 
342 of them survived


In [115]:
How many that survived were female and how many that died were female?

Object `female` not found.


In [116]:
df_f = df[df['Sex'] == 'female']
total_females = df_f.Sex.count()
f_survived = df_f.Survived.sum()
f_died = df_f.Survived.count() - df_f.Survived.sum()
check = total_females - f_survived - f_died == 0

print('There were ' + str(total_females) + ' on board, where '+ str(f_survived) + ' survived \nand ' + str(f_died) + ' died.')

check

There were 314 on board, where 233 survived 
and 81 died.


True

Q3: How many children were on the titanic?
Assumption that Children < 18

In [117]:
df_children = df[df['Age'] < 18]
numChildren = df_children.Age.count()

print('There were ' +  str(numChildren) + ' on board')

There were 113 on board


Q4: How many children died that were on the ship?

In [118]:
childSurvived = df_children.Survived.sum()
print('Out of the ' + str(numChildren) + ' Children that boarded Titanic, \n' + str(childSurvived) + ' survived')

Out of the 113 Children that boarded Titanic, 
61 survived


Q5 How many people had families with them?
Assumption: ParCh Sibsp mean family on board

In [119]:
Wfamily = df[(df['SibSp'] > 0) | (df['Parch'] > 0)]
numWfamily = Wfamily.PassengerId.count()

print('There were ' + str(numWfamily) + ' people with either parent,\nchildren, siblings or spouses on board')

There were 354 people with either parent,
children, siblings or spouses on board


Q6: What is the ratio of female to male?

In [120]:
total_females = df_f.Sex.count()
total = df.Sex.count()
sex_ratio_f = round((total_females / total) * 100, ndigits = 2)
sex_ratio_m = round(((total - total_females) / total) * 100, ndigits = 2)


print('The ratio of females initially on board was ' + str(sex_ratio_f) + '%')
print('The ratio of males initially on board was ' + str(sex_ratio_m) + '%')

The ratio of females on the board was 35.24%
The ratio of males on the board was 64.76%


Q7: What contributed to the survival of those who survived?
- 1st: See whether proportionally more f or m survived

In [121]:
df_S = df[df['Survived'] == 1]
df_f_S = df_S[df_S['Sex'] == 'female']
total_females_S = df_f_S.Sex.count()
total_S = df_S.Sex.count()
sex_ratio_f_S = round((total_females_S / total_S) * 100, ndigits = 2)
sex_ratio_m_S = round(((total_S - total_females_S) / total_S) * 100, ndigits = 2)

print('Of the survived people ' + str(sex_ratio_f_S) + '%' + ' were females.')

Of the survived people 68.13% were females.


As initially there were only 35% female on board, however 69% of the survivers were females, one could deduct that it woman might have been saved faster

- 2nd: Check whether childred or adults had a better chance of surviving

In [126]:
df_children = df[df['Age'] < 18]
CtoP_Ratio_before = round(df_children.Age.count() / total, ndigits= 2) *100

print('The ratio of children to adults initally on board was ' + str(CtoP_Ratio_before) + '%')

The ratio of children to adults on board was 13.0%


In [128]:
df_Child_S = df_S[df_S['Age'] < 18]
CtoP_Ratio_after = round(df_Child_S.Age.count() / df_S.Age.count(), ndigits= 2) *100

print('The ratio of children to adults initally on board was ' + str(CtoP_Ratio_after) + '%')

The ratio of children to adults initally on board was 21.0%


It seems that also Children (People w/ age < 18) had a bigger chance of survival. This becomes apparent as initally only 13% of the people on board were childrend. Of the surivivers there were however 21%. Hence, there is a clear increase in the percentage, and one could therefore deduct that childred had a higher chance of survival.